include/swift/Frontend/Frontend.h

```diff
@@ -404,7 +404,7 @@ class CompilerInstance {
   /// Null if no tracker.
   std::unique_ptr<DependencyTracker> DepTracker;
 
-  ModuleDecl *MainModule = nullptr;
+  mutable ModuleDecl *MainModule = nullptr;
```

Comment:
What is the reason for making this mutable? A comment would be worthwhile here.

Reply:
I hesitated on this one, but it's a singleton.  Its getter is logically `const`.  Will document this.